utils/node_local_test.py

```diff
@@ -1699,12 +1728,35 @@ def run_posix_tests(server, conf, test=None):
     """
 
     def _run_test():
-        start = time.time()
-        print('Calling {}'.format(fn))
-        rc = obj()
-        duration = time.time() - start
-        print('rc from {} is {}'.format(fn, rc))
-        print('Took {:.1f} seconds'.format(duration))
+        pt.call_index = 0
+        while True:
+            pt.needs_more = False
+            pt.test_name = fn
+            start = time.time()
+            print('Calling {}'.format(fn))
+            try:
+                pt.container = create_cont(conf, pool, posix=True)
+                rc = obj()
+                destroy_container(conf, pool, pt.container)
+                pt.container = None
+            except Exception as inst:
+                trace = ''.join(traceback.format_tb(inst.__traceback__))
+                duration = time.time() - start
+                conf.wf.add_test_case(pt.test_name,
+                                      repr(inst),
+                                      output = trace,
+                                      test_class='test',
+                                      duration = duration)
+                raise
```

Comment:
I don't understand what's going on with add_test_case here when there is an Exception.
What does `rc = obj()` do?

Reply:
Just before here there's the line **obj = getattr(pt, fn)** where pt is an object and fn is a string, this in effect gets a function pointer, so obj() is a method call into the object.  What this means is we can add new tests just by adding test_* methods to the object, which is exactly the same way unittest works.

Reply:
Oh, I didn't notice this `def _run_test():` was defiined within the scope of `def run_posix_tests`. This makes sense now

Reply:
yeah, it confused me what I saw it yesterday as well, I was wondering where all these locals came from!